### DISCUSSION TITLE
Try to modify Front End and add new custom functionality

### DISCUSSION THREAD

**Discussion by anagnostouJohn** [cea-sec/ivre#1112] [Q&A] [answered]
Hello i am new in bottle and i am currently working on personal project to add some custom functionalities to IVRE.
For instance i want to make a login page a  new pages for posting data etc. 
I am having difficulties in understanding how the WEB application is served.  For instance i have read how bottle serves pages but i cannot find where you serve the index.html (assuming is the main page) page in your application. How can i serve custom html pages?

**Comment by p-l-** [comment, marked as answer]
Static web files are served from `/usr/share/ivre/web/static/` (depending on where & how IVRE has been installed); this is either set in your Web server (Apache, Nginx, etc.) configuration, or, if you use the `ivre httpd` tool, by the `WEB_STATIC_PATH` configuration variable.

**Reply by anagnostouJohn** [reply]
Ok thanx for your reply. The way i understand is that ivre is an api that user calls it through search flow etc and then .js files renders the results. The `app.py `witch runs everything is located at `/usr/local/lib/python3.8/dist-packages/ivre/web/app.py.` Assuming that i have understand everything correct why when i add a new @application.get to make a custom call  lets say this line of codes
```
@application.get('/hello')
def hello():
    name = "john"
    return template('<b>Hello {{name}}</b>!', name=name)
```
I cannot get results nether from web nor from curl.
Can you help me somehow?

**Reply by p-l-** [reply]
Please format your code correctly, so that it can be read easily.

If you add your route for `/hello` in `ivre/web/app.py`, the app is "mounted" to `/cgi` by `ivre httpd`; so you would need to hit `/cgi/hello` so that your code gets called. Does that answer your question?

**Reply by anagnostouJohn** [reply]
YES YES YES million thank 2 last questions the httpd.py that is located at `/usr/local/lib/python3.8/dist-packages/ivre/tools` says that `runs a simple httpd server and used only for testing ` in your answer you mention that it is mounted to `/cgi `by `ivre httpd` i lost you here. Second i assume that you start a service in order to serve the api. could you mention it.  Thank you very match for your help i hope to not bother you any more.

**Reply by p-l-** [reply]
`ivre httpd` is a tool that you should not use to run a production server. It is only useful for local tests.

If you want to run a production server, you'll need to [install Nginx or Apache](https://doc.ivre.rocks/en/latest/install/installation.html#web). As stated in the documentation, you should refer to the web servers documentation on how to install them.

You can find quick "how-to" docs either in the [documentation](https://doc.ivre.rocks/en/latest/install/fast-install-and-first-run.html#install) or in the [web and web-apache Dockerfiles](https://github.com/cea-sec/ivre/tree/master/docker).

**Reply by anagnostouJohn** [reply]
thank you very much
